projects/components/src/table/data/table-data-source.ts

```diff
@@ -12,15 +12,27 @@ export interface TableDataRequest<TCol extends TableColumnConfig = TableColumnCo
     startIndex: number;
     limit: number;
   };
-  sort?: {
-    column: TCol;
-    direction: TableSortDirection;
-  };
+  sort?: SortSpecification<TCol>;
   filters?: TableFilter[];
   includeInactive?: boolean;
+
+  // New properties solely added for client side filtering specifications
+  clientSideFilters?: TableFilter[];
+  clientSideSort?: SortSpecification<TCol>;
 }
 
 export interface TableDataResponse<TData> {
   data: TData[];
   totalCount: number;
 }
+
+export interface SortSpecification<T> {
+  column: T;
+  direction: TableSortDirection;
+}
+
+// We need this information in the data source layer as the query needs to be reset to the original one to prevent refetch
```

Comment:
Doc comment so it shows up in VSCode Intellisense
```suggestion
/** 
 * We need this information in the data source layer as the query 
 * needs to be reset to the original one to prevent refetch
 */
```

Reply:
Cool, will do this.